lib/philomena/tag_changes.ex

```diff
@@ -69,6 +74,136 @@ defmodule Philomena.TagChanges do
   def full_revert(%{fingerprint: _fingerprint, attributes: _attributes} = params),
     do: Exq.enqueue(Exq, "indexing", TagChangeRevertWorker, [params])
 
+  @doc """
+  Updates tag change search indices when a user's name changes.
+
+  ## Examples
+
+      iex> user_name_reindex("old_username", "new_username")
+      :ok
+
+  """
+  def user_name_reindex(old_name, new_name) do
+    data = TagChangeIndex.user_name_update_by_query(old_name, new_name)
+
+    Search.update_by_query(TagChange, data.query, data.set_replacements, data.replacements)
+  end
+
+  @doc """
+  Queues a tag change for reindexing.
+
+  Adds the tag change to the indexing queue to update its search index.
+
+  ## Examples
+
+      iex> reindex_tag_change(tag_change)
+      %TagChange{}
+
+  """
+  def reindex_tag_change(%TagChange{} = tag_change) do
+    Exq.enqueue(Exq, "indexing", IndexWorker, ["TagChanges", "id", [tag_change.id]])
+
+    tag_change
+  end
+
+  @doc """
+  Queues all listed tag change IDs for search index updates.
+  Returns the list unchanged, for use in a pipeline.
+
+  ## Examples
+
+      iex> reindex_tag_changes([1, 2, 3])
+      [1, 2, 3]
+
+  """
+  def reindex_tag_changes(tag_change_ids) do
+    Exq.enqueue(Exq, "indexing", IndexWorker, ["TagChanges", "id", tag_change_ids])
+
+    tag_change_ids
+  end
+
+  @doc """
+  Queues all tag changes associated with a list of image IDs for search index updates.
+  Returns the list unchanged, for use in a pipeline.
+
+  ## Examples
+
+      iex> reindex_tag_changes_on_images([1, 2, 3])
+      [1, 2, 3]
+
+  """
+  def reindex_tag_changes_on_images(image_ids) do
+    Exq.enqueue(Exq, "indexing", IndexWorker, ["TagChanges", "image_id", image_ids])
+
+    image_ids
+  end
+
+  @doc """
+  Removes a tag change from the search index.
+
+  Deletes the tag change's document from the search index.
+
+  ## Examples
+
+      iex> unindex_tag_change(tag_change)
+      %TagChange{}
+
+  """
+  def unindex_tag_change(%TagChange{} = tag_change) do
+    Search.delete_document(tag_change.id, TagChange)
+
+    tag_change
+  end
+
+  @doc """
+  Returns a list of associations to preload when indexing tag changes.
+
+  ## Examples
+
+      iex> indexing_preloads()
+      [:image, :tags, :user]
+
+  """
+  def indexing_preloads do
+    alias_tags_query = select(Tag, [t], map(t, [:aliased_tag_id, :name]))
+
+    base_tags_query =
+      Tag
+      |> select([t], [:category, :id, :name])
+      |> preload(aliases: ^alias_tags_query)
+
+    image_query =
+      Image
+      |> select([i], struct(i, [:anonymous, :user_id]))
+
+    [
+      image: image_query,
+      tags: [
+        tag: base_tags_query
+      ],
+      user: select(User, [u], map(u, [:id, :name]))
```

Comment:
```suggestion
      user: select(User, [:name])
```

Reply:
(on this and 2 other similar comments) then this should be refactored in the other searchable things, because I kind of copy-pasted some of this code from them.